proxy-libs/heliossarama/sarama_test.go

```diff
@@ -0,0 +1,186 @@
+package heliossarama
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/Shopify/sarama"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	"go.opentelemetry.io/otel/trace"
+	exportsExtractor "github.com/helios/go-instrumentor/exports_extractor"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+)
+
+var (
+	Addresses []string = []string{"localhost:9093"}
+	Topic     string   = "testTopic"
+)
+
+type TestConsumerGroupHandler struct {
+	messageKey   string
+	messageValue string
+	spanRecorder *tracetest.SpanRecorder
+	t            *testing.T
+}
+
+func (handler *TestConsumerGroupHandler) Setup(session ConsumerGroupSession) error {
+	return nil
+}
+
+func (handler *TestConsumerGroupHandler) Cleanup(session ConsumerGroupSession) error {
+	return nil
+}
+
+func (handler *TestConsumerGroupHandler) ConsumeClaim(session ConsumerGroupSession, claim ConsumerGroupClaim) error {
+	message := <-claim.Messages()
```

Comment:
Can you create a span here and check that its parent is the consume span?

Reply:
Same problem with the context propagation here - I have to manually extract the context from the message (with `otelsarama`), and pass it on, in the test itself - So I wouldn't be checking anything really.